example/thirdparty/ollama-js/build.mill

```diff
@@ -0,0 +1,318 @@
+package build
+
+import mill._, javascriptlib._
+import os.*
+import ujson._
+
+trait OllamaModule extends TypeScriptModule {
+  // bundle ollama with `unbuild`
+  def unbuild = Task {
+    createNodeModulesSymlink()
+    val out = compile().path
+
+    os.walk(out, skip = p => p.last == "node_modules" || p.last == "package-lock.json")
+      .foreach(p => os.copy.over(p, T.dest / p.relativeTo(out), createFolders = true))
+
+    os.call("node_modules/.bin/unbuild", cwd = T.dest)
+
+    // prepare bundled ouptut to be used as `unmanagedDeps` in examples section.
+    os.makeDir.all(T.dest / "ollama")
+    os.copy(T.dest / "package.json", T.dest / "ollama" / "package.json")
+    os.copy(T.dest / "dist", T.dest / "ollama" / "dist")
+
+    PathRef(T.dest / "ollama")
+  }
+}
+
+trait ExampleModule extends TypeScriptModule {
+  def sources = Task.Sources(moduleDir)
+  def mainFilePath = Task { compile().path / mainFileName() }
+}
+
+trait EsmModule extends ExampleModule {
+  def enableEsm = true
+}
+
+trait Es6Module extends ExampleModule {
+  def compilerOptions =
+    super.compilerOptions() + (
+      "module" -> ujson.Str("nodenext"),
+      "moduleResolution" -> ujson.Str("nodenext"),
+      "target" -> ujson.Str("es6")
+    )
+}
+
+object `package` extends RootModule with OllamaModule with TsLintModule {
+  def moduleName = "core"
+  def npmDeps = Seq("whatwg-fetch@3.6.20")
+  def npmDevDeps = Seq(
+    "@swc/core@1.10.12",
+    "@types/whatwg-fetch@0.0.33",
+    "unbuild@2.0.0"
+  )
+
+  def npmLintDeps = Seq(
+    "prettier@3.2.4",
+    "eslint@8.29.0",
+    "@typescript-eslint/eslint-plugin@5.42.1",
+    "@typescript-eslint/parser@5.42.1"
+  )
+
+  // disable declarations output in compile stage
+  def runTypeCheck: T[Boolean] = false
+
+  object test extends TypeScriptTests with TestModule.Vitest
+
+  object examples extends TypeScriptModule {
+    object abort extends ExampleModule {
+      def unmanagedDeps = Seq(unbuild())
+      def mainFileName = s"abort-all-requests.ts"
+    }
+
+    object `fill-in-middle` extends EsmModule {
+      def unmanagedDeps = Seq(unbuild())
```

Comment:
Can this `def managedDeps` declaration also be moves to `ExampleModule`?

Reply:
nope.

Reply:
y?

Reply:
unbuild() is a custom task defined within `package`, it exists only within the scope of `package`.
There is no way to access it from `EsmModule` that would get rid of the multiple iterations of unbuild() being called in each `EsmModule` object.

Reply:
actually yes we can, making the changes

Reply:
done.

Reply:
@lihaoyi